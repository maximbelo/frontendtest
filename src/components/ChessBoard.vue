<template>
  <div class="chessboard">
    <!-- Render ranks (rows) -->
    <div v-for="rank in ranks" :key="rank" class="rank">
      <!-- Render files (cols) inside the ranks -->
      <div
        v-for="file in files"
        :key="file"
        :class="getSquareClass(file, rank)"
        @click="selectSquare(file, rank)"
      ></div>
    </div>
  </div>
  <!-- ./chessboard -->
</template>

<script setup lang="ts">
import { useSelectedSquareStore } from "@/stores/selectedSquare";

const store = useSelectedSquareStore();

const files = ["a", "b", "c", "d", "e", "f", "g", "h"];
const ranks = ["1", "2", "3", "4", "5", "6", "7", "8"].reverse();

const selectSquare = (file: string, rank: string): void => {
  // When a user clicks on the square, this calls the selectSquare action in the Pinia store to update the selected square state.
  store.selectSquare(file, rank);
};

const getSquareClass = (file: string, rank: string): string => {
  // Determine if the square is a light square based on its position on the board.
  const isLightSquare = (files.indexOf(file) + ranks.indexOf(rank)) % 2 === 0;

  // Return the appropriate class based on whether it's a light or dark square.
  return isLightSquare ? "square dark" : "square light";
};
</script>

<style lang="scss" scoped>
.chessboard {
  width: 100%;
}

.rank {
  display: flex;
}

.square {
  width: 70px;
  height: 70px;
}

.square.dark {
  background-color: #b58863;
}

.square.light {
  background-color: #f0d9b5;
}
</style>
