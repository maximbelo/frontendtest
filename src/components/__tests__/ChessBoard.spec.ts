import { describe, expect, test, beforeEach } from "vitest";
import { shallowMount } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import ChessBoard from "@/components/Chessboard.vue";

describe("ChessBoard.vue tests", () => {
  beforeEach(() => {
    // create a fresh Pinia instance and make it active so it's automatically picked
    // up by any useStore() call without having to pass it to it:
    // `useStore(pinia)`
    setActivePinia(createPinia());
  });

  test("matches snapshot", () => {
    const wrapper = shallowMount(ChessBoard);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("renders the correct number of squares", () => {
    const wrapper = shallowMount(ChessBoard);
    const squares = wrapper.findAll(".square");
    expect(squares).toHaveLength(64);
  });

  test("highlights the selected square on click", async () => {
    const wrapper = shallowMount(ChessBoard);
    const firstSquare = wrapper.find(".square");

    await firstSquare.trigger("click");

    expect(firstSquare.classes()).toContain("highlighted");
  });
});
