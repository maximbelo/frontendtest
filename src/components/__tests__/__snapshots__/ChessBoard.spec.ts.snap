// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChessBoard.vue tests > matches snapshot 1`] = `
"<div data-v-41886935=\\"\\" class=\\"chessboard\\">
  <div data-v-41886935=\\"\\" class=\\"rank\\">
    <!-- Render ranks (rows) -->
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <p data-v-41886935=\\"\\" class=\\"files-display\\">8</p><!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
  </div>
  <div data-v-41886935=\\"\\" class=\\"rank\\">
    <!-- Render ranks (rows) -->
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <p data-v-41886935=\\"\\" class=\\"files-display\\">7</p><!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
  </div>
  <div data-v-41886935=\\"\\" class=\\"rank\\">
    <!-- Render ranks (rows) -->
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <p data-v-41886935=\\"\\" class=\\"files-display\\">6</p><!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
  </div>
  <div data-v-41886935=\\"\\" class=\\"rank\\">
    <!-- Render ranks (rows) -->
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <p data-v-41886935=\\"\\" class=\\"files-display\\">5</p><!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
  </div>
  <div data-v-41886935=\\"\\" class=\\"rank\\">
    <!-- Render ranks (rows) -->
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <p data-v-41886935=\\"\\" class=\\"files-display\\">4</p><!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
  </div>
  <div data-v-41886935=\\"\\" class=\\"rank\\">
    <!-- Render ranks (rows) -->
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <p data-v-41886935=\\"\\" class=\\"files-display\\">3</p><!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
  </div>
  <div data-v-41886935=\\"\\" class=\\"rank\\">
    <!-- Render ranks (rows) -->
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <p data-v-41886935=\\"\\" class=\\"files-display\\">2</p><!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <!--v-if-->
      <!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
  </div>
  <div data-v-41886935=\\"\\" class=\\"rank\\">
    <!-- Render ranks (rows) -->
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <p data-v-41886935=\\"\\" class=\\"ranks-display\\">a</p><!-- Display the ranks on the bottom squares -->
      <p data-v-41886935=\\"\\" class=\\"files-display\\">1</p><!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <p data-v-41886935=\\"\\" class=\\"ranks-display\\">b</p><!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <p data-v-41886935=\\"\\" class=\\"ranks-display\\">c</p><!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <p data-v-41886935=\\"\\" class=\\"ranks-display\\">d</p><!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <p data-v-41886935=\\"\\" class=\\"ranks-display\\">e</p><!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <p data-v-41886935=\\"\\" class=\\"ranks-display\\">f</p><!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square dark\\">
      <!-- Render files (cols) inside the ranks -->
      <p data-v-41886935=\\"\\" class=\\"ranks-display\\">g</p><!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
    <div data-v-41886935=\\"\\" class=\\"square light\\">
      <!-- Render files (cols) inside the ranks -->
      <p data-v-41886935=\\"\\" class=\\"ranks-display\\">h</p><!-- Display the ranks on the bottom squares -->
      <!--v-if-->
      <!-- Display the files on the far left squares -->
    </div>
  </div>
</div>
<!-- ./chessboard -->"
`;
